src/lib/db/suggest-change-store.ts

```diff
@@ -0,0 +1,123 @@
+import { ISuggestChangeStore } from '../types/stores/suggest-change-store';
+import { Logger, LogProvider } from '../logger';
+import EventEmitter from 'events';
+import { Knex } from 'knex';
+import { PartialSome } from '../types/partial';
+import { ISuggestChange, ISuggestChangeSet } from '../types/model';
+import User from '../types/user';
+import NotFoundError from '../error/notfound-error';
+
+const Tables = {
+    suggestChangeSet: 'suggest_change_set',
+};
+
+const COLUMNS = [
+    'id',
+    'environment',
+    'state',
+    'project',
+    'changes',
+    'created_by',
+    'created_at',
+    'updated_by',
+];
+
+interface ISuggestChangeSetRow {
+    id: number;
+    environment: string;
+    state?: string;
+    project?: string;
+    created_by?: string;
+    created_at?: Date;
+    changes?: ISuggestChange[];
+}
+
+export default class SuggestChangeStore implements ISuggestChangeStore {
+    private logger: Logger;
+
+    private eventBus: EventEmitter;
+
+    private db: Knex;
+
+    constructor(db: Knex, eventBus: EventEmitter, getLogger: LogProvider) {
+        this.db = db;
+        this.eventBus = eventBus;
+        this.logger = getLogger('lib/db/suggest-change-store.ts');
+    }
+
+    async create(
+        suggestChangeSet: PartialSome<ISuggestChangeSet, 'id'>,
+        user: Partial<Pick<User, 'username' | 'email'>>,
+    ): Promise<ISuggestChangeSet> {
+        const rows = await this.db(Tables.suggestChangeSet)
+            .insert<ISuggestChangeSetRow>({
+                id: suggestChangeSet.id,
+                environment: suggestChangeSet.environment,
+                state: suggestChangeSet.state,
+                project: suggestChangeSet.project,
+                changes: JSON.stringify(suggestChangeSet.changes),
+                created_at: suggestChangeSet.createdAt,
+                created_by: user.username || user.email,
+            })
+            .returning(COLUMNS);
+
+        return this.mapRow(rows[0]);
+    }
+
+    async getAll(): Promise<ISuggestChangeSet[]> {
+        const rows: ISuggestChangeSet[] = await this.db
+            .select(this.prefixColumns())
+            .from(Tables.suggestChangeSet)
+            .orderBy('id', 'asc');
+
+        return rows.map(this.mapRow);
+    }
+
+    async get(id: number): Promise<ISuggestChangeSet> {
+        const rows: ISuggestChangeSetRow[] = await this.db
+            .select(this.prefixColumns())
+            .from(Tables.suggestChangeSet)
+            .where({ id });
+
+        return this.mapRow(rows[0]);
+    }
+
+    prefixColumns(): string[] {
+        return COLUMNS.map((c) => `${Tables.suggestChangeSet}.${c}`);
+    }
+
+    mapRow(row?: ISuggestChangeSetRow): ISuggestChangeSet {
+        if (!row) {
+            throw new NotFoundError('No row');
+        }
+
```

Comment:
Out of curiosity: How do we usually handle cases like this? I'm assuming you're following the same conventions that we use elsewhere in the code base, so it's probably perfectly on brand.

On a personal level, though, I prefer validating values before they're being sent into functions, so that function don't have to check for invalid arguments. So the function would be like this instead:

```suggestion
    mapRow(row: ISuggestChangeSetRow): ISuggestChangeSet {
```

That said, this is more to facilitate a discussion than to point to something we need to change. I also think there's value in keeping with patterns that we use elsewhere in the code, so there's that 💁🏼